kernel/nothread.c

```diff
@@ -37,12 +37,20 @@ int32_t z_impl_k_sleep(k_timeout_t timeout)
 
 	ticks = timeout.ticks;
 	if (Z_TICK_ABS(ticks) <= 0) {
-		expected_wakeup_ticks = ticks + sys_clock_tick_get_32();
+		/* ticks is delta timeout */
+		ticks_to_wait = ticks;
 	} else {
-		expected_wakeup_ticks = Z_TICK_ABS(ticks);
+		/* ticks is absolute timeout expiration */
+		uint32_t curr_ticks = sys_clock_tick_get_32();
+
+		if (Z_TICK_ABS(ticks) > curr_ticks) {
+			ticks_to_wait = Z_TICK_ABS(ticks) - curr_ticks;
```

Comment:
Are they possible to be negative number?

(Just a question.)

Reply:
Not really understand question.

If you talking about `Z_TICK_ABS(ticks)`, than it can. But negative values are caught by previous if.
If you talking about `ticks_to_wait`, I think there is no meaning in waiting negative time. So, I make it unsigned.
`sys_clock_tick_get_32()`, of course, can not be negative.